src/frontend/screens/Observation/index.tsx

```diff
@@ -57,9 +61,15 @@ export const ObservationScreen: NativeNavigationComponent<'Observation'> = ({
       }, defaultAcc)
     : [];
 
-  const deviceId = '';
   const {lat, lon, createdBy} = observation;
-  const isMine = deviceId === createdBy;
+  const {data: deviceInfo, isPending: isDeviceInfoPending} = useDeviceInfo();
+  const {deviceId} = deviceInfo || {};
+  const {data: convertedDeviceId, isPending: isCreatedByDeviceIdPending} =
+    useCreatedByToDeviceId(createdBy);
+  const isMine =
+    !isDeviceInfoPending &&
+    !isCreatedByDeviceIdPending &&
+    deviceId === convertedDeviceId;
```

Comment:
nit: ideally this page would make use of the query statuses to wait before rendering relevant parts or passing data to components. for example, don't render the `ButtonFields` until data from both queries is available.

aside from that, i think this check can be more explicit to make sure that `deviceId` and `convertedDeviceId` are not undefined (at the TS level, at least):

```suggestion
  const {data: deviceInfo} = useDeviceInfo();
  const {data: convertedDeviceId} = useCreatedByToDeviceId(createdBy);
  const isMine =
    deviceInfo?.deviceId !== undefined &&
    convertedDeviceId !== undefined &&
    deviceInfo.deviceId === convertedDeviceId;
```

Reply:
The only thing I found to conditionally render was the button fields, which I did. Is that what you had in mind? Were there other parts or components that were not being handled as you wished?

Reply:
that was the only one that seemed relevant, thanks!